### DISCUSSION TITLE
Canvas render filled path slowly

### DISCUSSION THREAD

**Discussion by GISerliang** [hecrj/iced#1293] [Q&A]
I used `frame.fill` function to render some polygon, but I found that was slowly. The speend times like this:

```
1235 points
render speed 6.6532ms
1560 points
render speed 9.6309ms
1665 points
render speed 14.5301ms
1252 points
render speed 8.8198ms
1609 points
render speed 13.1612ms
3012 points
render speed 56.7091ms
1819points
render speed 22.8558ms
2127 points
render speed 35.1137ms
1793 points
render speed 15.5816ms
1143 points
render speed 8.2164ms

```

Is there a faster way to render polygon? Thank you.

**Comment by hecrj** [comment]
What are you measuring exactly? What is "render speed"?

**Reply by GISerliang** [reply]
> What are you measuring exactly? What is "render speed"?

If I added a lot of data to canvas once and needed wait long time.  
I used `frame.stroke` function faster than  `frame.fill`, why the two functions renderer was difference so much ?
And I could not render the outline of polygon.

**Reply by hecrj** [reply]
But what are you exactly measuring? Frame time? Function calls? Can you provide some code to reproduce your results?

Also, are you using `--release` mode?

**Reply by GISerliang** [reply]
No, I not using `--release` mode, only `debug`.

**Reply by hecrj** [reply]
Performance in `debug` mode is not indicative of actual `release` performance.

**Reply by GISerliang** [reply]
Ok, I will try.
